tests/promising_function/test_promising_function.py

```
@@ -0,0 +1,621 @@
+async def test_exception_in_class_init() -> None:
+    """
+    An exception in __init__ raises synchronously
+    (before Promise creation).
+
+    TODO We might want to deviate from this later, though.
+    """
+
+    @promising.function
+    class FailingInit:
+        def __init__(self) -> None:
+            raise TypeError("init error")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)